invenio_communities/members/services/components.py

```diff
@@ -50,20 +47,31 @@ def _get_user_ids_from_group_id(role_id):
             )
 
         for user_id in user_ids:
-            on_membership_change(Identity(user_id))
+            on_user_membership_change(Identity(user_id))
 
     def accept_invite(self, identity, record=None, data=None, **kwargs):
         """On accept invite."""
         self._member_changed(record)
 
-    def members_add(self, identity, record=None, data=None, **kwargs):
+    def members_add(self, identity, record=None, community=None, data=None, **kwargs):
         """On member add (only for groups)."""
-        self._member_changed(record)
+        if record.get("type") == "group":
+            role = Role.query.filter_by(id=record["id"]).one_or_none()
+            if role.is_managed:
+                users = role.users.all()
+                for user in users:
+                    on_user_membership_change(Identity(user.id))
```

Comment:
Here there is room for improvement:
- we get the user id
- we wrap it around an Identity
- then, when deleting, we get again the `id`